### DISCUSSION TITLE
[Book 1, Chapter 10] Shadows and reflections do not appear after adding the material class.

### DISCUSSION THREAD

**Discussion by sapientsand** [RayTracing/raytracing.github.io#1636] [Debugging]
Hello. I'm not sure what I'm doing wrong, but after I added  the material class to my program, there aren't any shadows in my output images. I'm not exactly sure what I'm missing here, but I'll attach image and a link to my code below to see if someone more competent than I am can figure out what I've done wrong. My code in question: https://github.com/sapientsand/rtinoneweekend

![Output Image](https://github.com/user-attachments/assets/fb8fa5cd-6653-45ce-8cff-0cd709db80ec)

**Comment by whydoubt** [comment]
I suspect there may be more differences, but at the least, in the `metal` class, the line
```C++
vec3 reflected = unit_vector(reflected) + (fuzz * random_unit_vector());
```
does not match the book.
